### PR TITLE
Feature/integrate op plugins plugin

### PR DESCRIPTION
**Pull request by myabc**
https://www.openproject.org/work_packages/14305

Relates to: https://www.openproject.org/work_packages/14004

### CODE REVIEW

File: lib/open_project/plugins/acts_as_op_engine.rb [hunk 1]
```diff
@@ -0,0 +1,159 @@
+#-- copyright
+# OpenProject Plugins Plugin
```
Comment (tessi) [inline lib/open_project/plugins/acts_as_op_engine.rb]: I'm afraid you have to adjust the copyright to fit into the OP core.

Comment (myabc) [inline lib/open_project/plugins/acts_as_op_engine.rb]: @tessi good point!

Comment (myabc) [inline lib/open_project/plugins/acts_as_op_engine.rb]: Fixed in bba0c49d.

**Comment by ulferts** [comment]
This breaks because the dependencies are no longer resolved like they used to be.

As Bundler.require is called very early in the boot process, and this triggers all the plugins to be loaded,  the OpenProject::Plugins::Engine needs to be loaded before that. But because the lib directory is not yet part of the autoloading at this point, we will need to require all of that by hand. It is of course doable but will require some extra work.

**Pull request merged**